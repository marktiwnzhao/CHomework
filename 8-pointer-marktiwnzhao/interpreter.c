/**
 * File: interpreter.c
 *
 * Author:zhaozhengjie
 * ID:211250109
 * Date:2021.12.6
 */
#include <stdio.h>
#include <stdlib.h>
#define LEN 10
int main() {
    char *str = malloc(LEN * sizeof(*str));
    scanf("%s", str);
    int num = 0;
    int *res = &num;
    sscanf(str, "%x", res);
    printf("%ld\n%lu\n", *res, *res);
    float *p = (float *)res;
    printf("%f", *p);
    free(str);
    return 0;
}

