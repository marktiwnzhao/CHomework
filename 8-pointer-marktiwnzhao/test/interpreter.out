-2147483622
2147483674
-0.000000